### ISSUE TITLE
xray/vless/xtls/trojan all speed limited as of today

### ISSUE THREAD

**Issue by masoudelete** [net4people/bbs#185]
I have a server which I am it's only user. 
I made it using https://github.com/XTLS/Xray-core oneclick scripts with a fake website.

it worked very well on all isp's but as of today all of my protocols(vless xtls, trojan,...) are throttled to 10 or 20kb/s and I was just wondering why that is.

is it possible that they have found a way to fingerprint these protocols? is my IP got greylisted or is it just bulk throttling a range of ips

if you have a similar problem please share
and if there is a better protocol than xray-core please let me know
i'm ready to share any test or log file

**Comment by arandomgstring** [comment]
@Azadzadeh 

My two cents (that you may like to add to your website):

1. Not receiving PING from an IP doesn't always indicate a blockage. Many Windows Servers by default block ping requests. Therefore, even a clean healthy IP might respond with timeouts only. If the VPS owner can connect to server via SSH or RDC or whatever, that means the IP itself is not blocked, rather, firewall doesn't let ping requests to go through. 

2. SSH too works fine. But port 22 isn't pretty. To use SSH, one can simply run a command like `ssh -D xxxx user@host -p yyyy` where xxxx is a local port on user's device which is 100% arbitrary and yyyy is ssh port of server. Then with a proxifier (or any other application) you can tunnel all requests to xxxx which will be redirected to host:yyyy. I have tested this, and it works like a charm for now. Http Injector on android can be used for a similar purpose. On IOS it seems everything needs payment (dah) but perhaps termux can be used. Haven't tested this, I am poor and don't have IPhone. 

And finally I think Iranians need a real forum for these type of discussions. Github is fine, but I don't want it to be blocked. Maybe an onion site would be nice. It can even contain a market for proxies, so that developers sell what they have made with crypto or whatever.

**Comment by arandomgstring** [comment]
@Azadzadeh 

> have you seen a situation in iran where the ping failed but the IP was accessible (and the user was not under some local firewall)?

Of course. That's why I have mentioned this anyway. Note that I didn't mean user's (or ISP's) firewall, rather I meant the proxy server's firewall. Your windows server's firewall might be configured to drop ICMP requests. 

> I mentioned this in [here](https://azadzadeh.github.io/iran-internet/docs/guides/unix-redirect-apps/). Or did you mean creating a tun device so all programs would be redirected by default?

I don't see anything related to SSH though.

**Comment by Hadi-1624** [comment]
@arandomgstring I have a problem with using  ssh as a proxy
Youtube will not load, any ideas why this is?
i'm using windows 10's openssh client, i bind it to port 9999 and i use proxifier.

I hope an onion site pops up for these sorts of discussions, we really do need it.

**Comment by arandomgstring** [comment]
@Hadi-1624 
If other websites works (and you see your IP has changed) then it's DNS problem. In proxifier, click on name resolution, and check resolve hostnames through proxy.

**Comment by liulilittle** [comment]
> I have a server which I am it's only user. I made it using https://github.com/XTLS/Xray-core oneclick scripts with a fake website.
> 
> it worked very well on all isp's but as of today all of my protocols(vless xtls, trojan,...) are throttled to 10 or 20kb/s and I was just wondering why that is.
> 
> is it possible that they have found a way to fingerprint these protocols? is my IP got greylisted or is it just bulk throttling a range of ips
> 
> if you have a similar problem please share and if there is a better protocol than xray-core please let me know i'm ready to share any test or log file

You can use various tools, such as speedtest-cli, iperf, and so on,  to verify whether the rate is limited to the entire machine IP or only to some TCP application layer protocols.

If it's just speed limiting for specific the SSL/TLS/HTTP protocols, then there are many solutions,  you can use shadowsocks, uds, frp, ppp tools, etc.

Simple solution:
1. Speed limiting HTTP traffic (excluding HTTPS traffic) can be changed to HTTP traffic

2. Use other TCP application-layer proxy protocols that support encryption, such as uds, frp, and shadowsocks
